py/selenium/webdriver/common/action_chains.py

```diff
@@ -27,6 +31,11 @@
 from .actions.wheel_input import WheelInput
 from .utils import keys_to_typing
 
+any_device = Union[KeyInput, PointerInput, WheelInput]
+
+if TYPE_CHECKING:
+    from typing import Self
```

Comment:
Self is a recent feature, so you may need to update the dependency to include `typing_extension`
Also put all the other types in here.
and maybe put `from __future__ import annotations`.

Reply:
Recent feature as in not applicable to Python 3.8 which we still have to support?

Reply:
@titusfortner Python 3.8 doesn't have `Self` in the `typing` module but we should probably generalize the typing code so that it just uses typing_extensions by default. Probably not a bad idea really since typing extensions can be updated during a Python version's lifetime (like at this time, it could have fixes for `Self` that's not in Python 3.11.

Reply:
Ah, so pull it in as a dependency instead of using the one in the Python standard library?

Reply:
As far as I know it's not a bad practice.

Reply:
in general we try to minimize dependencies, but it seems like a good way to provide more flexibility.

Reply:
> Self is a recent feature, so you may need to update the dependency to include `typing_extension`
> Also put all the other types in here.
> and maybe put `from __future__ import annotations`.

Instead of `Self` I can used `TypeVar` as well. I separate out the type hints in this PR and raise a new one.